File: Top 50 Coding Interview Questions/03- Write code of Greatest Common Divisor/GCD.cpp
#include <iostream>
using namespace std;

// Function to calculate GCD using the Euclidean algorithm
int findGCD(int a, int b) {
    // Ensure the inputs are non-negative
    if (a < 0) a = -a;
    if (b < 0) b = -b;

    // Loop until one of the numbers becomes zero
    while (b != 0) {
        int remainder = a % b; // Find the remainder of a divided by b
        a = b;                 // Assign b to a
        b = remainder;         // Assign remainder to b
    }
    return a; // GCD is the last non-zero value of a
}

int main() {
    int num1, num2;

    // Input two integers from the user
    cout << "Enter two integers: ";
    cin >> num1 >> num2;

    // Calculate and display the GCD
    cout << "GCD of " << num1 << " and " << num2 << " is " << findGCD(num1, num2) << endl;

    return 0;
}

File: Top 50 Coding Interview Questions/03- Write code of Greatest Common Divisor/GCD.py
# Function to calculate GCD using the Euclidean algorithm
def find_gcd(a, b):
    """
    Calculates the Greatest Common Divisor (GCD) of two numbers using the Euclidean algorithm.
    """
    # Ensure the inputs are non-negative
    a, b = abs(a), abs(b)

    # Loop until one of the numbers becomes zero
    while b != 0:
        a, b = b, a % b  # Update a with b and b with the remainder

    return a  # GCD is the last non-zero value of a

# Input two integers from the user
num1 = int(input("Enter the first integer: "))
num2 = int(input("Enter the second integer: "))

# Calculate and display the GCD
print(f"GCD of {num1} and {num2} is {find_gcd(num1, num2)}")
